### DISCUSSION TITLE
I'd like to contribute

### DISCUSSION THREAD

**Discussion by tygoee** [Neko-Nik-Org/Auth-N#3] [General]
I say you on reddit a few days ago and starred the repo to come back, but I have some time and would like to help if you need any :)

I have about 2 years of (personal) python experience and am kinda bored without nice ideas

Just reply here if you want me to do anything

**Comment by Neko-Nik** [comment]
@tygoee Apologies for the delayed response; I was occupied with tasks for my company and couldn't reply earlier. Let's discuss our objectives:

Our primary goal is to reach a stage where this can be integrated into applications. I'm currently working on setting up OAuth 2.0 and its various flows: Authorization Code, PKCE, Implicit, Device Code, and OpenID Connect. You can find more information and a playground for testing the flows at https://www.oauth.com/

While we already have an access token in place, we need a flow that enables us to generate and handle refresh tokens as well. Please explore and implement any suitable types of flows for this purpose

The filenames or endpoint names might not be accurate at this stage, but our initial focus is on functionality. Once we achieve that, we can present it and seek funding

Thanks for your interest in my project. Your support and input are greatly appreciated

**Comment by tygoee** [comment]
Thanks for the reply, I'm reading some docs on oauth but I'll have some time this weekend to look at it. I kind of understand the process now but implementing it is probably harder. I think I'm gonna try implementing it into a website I'm currently developing too (which will probably never reach the public lol)

**Reply by Neko-Nik** [reply]
Ok, let's have a virtual meet when you are available and we can discuss about oauth and other things

**Reply by Neko-Nik** [reply]
Schedule a meeting with me at [Calendly - Nikhil](https://calendly.com/neko-nik/general-meet)

**Reply by tygoee** [reply]
Can we also discuss it over chat? I don't really want to do it over video or microphone, we can just keep it in this thread or you can add me somewhere else.

**Reply by Neko-Nik** [reply]
Okay, sure. Let's not have a video call if you're uncomfortable with it. (I suggested meet because its faster to explain)
My aim is to complete the basic OAuth 2.0 Flow. Currently, we have yet to implement the Refresh token part. I need support to develop this part, you can handle that part. We need the "refresh token to access token" generator endpoint, and we also need to think about how we can store and generate the token more efficiently and quickly, preferably stateless.
I can't wrap my design around it, so it would be really helpful if you could take over this part. With this part done, we will have a simple JWT token-based auth with a refresh mechanism. Once that is done, we can work on the main flow as mentioned earlier: Authorization Code, PKCE, Implicit, Device Code, and OpenID Connect. Thanks for your support
